modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3BlobContainer.java

```diff
@@ -537,7 +538,28 @@ String buildKey(String blobName) {
     }
 
     /**
-     * Uploads a blob using a single upload request
+     * Enumeration of mutually exlusive conditional operations supported by S3.
+     *
+     * @see <a href=https://docs.aws.amazon.com/AmazonS3/latest/userguide/conditional-requests.html>S3-conditional-requests</a>
+     */
+    sealed interface ConditionalOperation permits ConditionalOperation.IfMatch, ConditionalOperation.IfNoneMatch,
```

Comment:
Makes sense to me - let's pull this out to a preliminary PR, and also apply it to the complete-MPU location.